### PR TITLE
Update Steps in README to use Docker Hub

### PR DESCRIPTION
**Pull request by ajbozarth**
Given the model is available on Docker Hub we should update the Steps in the README.

This means less code for the user to checkout locally and make more sense in this usecase.

This update is also in line with updating the models with similar changes.

### CODE REVIEW

**Comment by ajbozarth** [comment]
This pr is a first step in the process of updating our model readme's to include steps to use docker hub. Daniel and I have already done some back and forth on this but we would like more feedback on wording and clarity of steps. (As well and quickly testing this works for everyone)

File: README.md [hunk 1]
```diff
@@ -161,7 +147,7 @@ Use the `model/predict` endpoint to load a test file and get captions for the im
 
 You can also test it on the command line, for example:
 
-    curl -F "image=@assets/surfing.jpg" -X POST http://localhost:5000/model/predict
+    curl -F "image=@path/to/image.jpg" -X POST http://localhost:5000/model/predict
```
Comment (MLnick) [inline README.md]: My only thought here is that, since the model repo was not checked out, the test assets are not readily available. But we still have the JSON response for one of them below.

Perhaps we can say "Use the `model/predict` endpoint to get captions for an image from the API. The `[model assets folder](link)` contains a number of images you can use to test out the API."

Or something like that?

File: README.md [hunk 2]
```diff
@@ -232,7 +218,7 @@ network stack. This is done in the following steps:
 Modify the command that runs the Image Caption Generator REST endpoint to map an additional port in the container to a
 port on the host machine. In the example below it is mapped to port `8088` on the host but other ports can also be used.
 
-    docker run -it -p 5000:5000 -p 8088:8088 --name max-im2txt max-im2txt
+    docker run -it -p 5000:5000 -p 8088:8088 --name max-im2txt codait/max-image-caption-generator
```
Comment (MLnick) [inline README.md]: While we are changing this, I wonder if the `--name` should also change? im2txt was more a WIP name before publishing, so perhaps we should change that to `max-image-caption-generator` (even though a bit more verbose)

File: README.md [hunk 3]
```diff
@@ -242,6 +228,14 @@ Run the web app container using:
 
     docker run --net='container:max-im2txt' -it webapp
 
+##### Using the Docker Hub Image
+
+You can also deploy the web app with the latest docker image available on DockerHub by running:
+
+    docker run --net='container:max-im2txt' -it codait/max-image-caption-generator-web-app
```
Comment (MLnick) [inline README.md]: Same comment applies here, changing `max-im2txt` to `max-image-caption-generator`

File: README.md [hunk 4]
```diff
@@ -242,6 +228,14 @@ Run the web app container using:
 
     docker run --net='container:max-im2txt' -it webapp
```
Comment (MLnick) [inline README.md]: Same comment applies here, changing `max-im2txt` to `max-image-caption-generator`

File: README.md (hunk 1 above)
Comment (ajbozarth) [reply README.md]: That's why I changed it to an obviously fake path instead of the asset that isn't there anymore. @djalova and I discussed this over slack and we didn't want to remove the "example" part from the code pattern since it's a landing point for more than just MAX.

File: README.md (hunk 2 above)
Comment (ajbozarth) [reply README.md]: I would agree here and I can update this for the code pattern, but this is how all the model repos have their naming, based on the original model name. We originally just copied this from the model repo, so should we update all the model repos as well? I could do it while adding the docker hub steps when I open those prs

**Comment by MLnick** [comment]
Yeah I’d say we should update the old docker image tags similarly - can do
when we add hub steps as you suggest

**Comment by ajbozarth** [comment]
I updated the docker container names as suggested

File: README.md (hunk 1 above)
Comment (MLnick) [reply README.md]: Sure - I think it just ties the example output to a specific example input and we could just mention the model repo as a location of some useful test image.. But not a huge deal either way.

Comment (ajbozarth) [reply README.md]: @stevemart @rhagarty in terms of this as a code pattern what do you guys think?

Comment (stevemart) [reply README.md]: @ajbozarth i am OK with the obviously fake path, but i would suggest adding a comment under the curl command, something like ...

> See https://github.com/IBM/MAX-Image-Caption-Generator/tree/master/assets for sample images to test with, or supply your own

Comment (ajbozarth) [reply README.md]: Thanks @stevemart  I'll add that line, but do you think we should leave in the following example output still?

Comment (ajbozarth) [reply README.md]: @MLnick @stevemart I added a clarifying section with a link to the asset folder based on your text suggestions if you could check it. After adding it though I'm unsure whether we should remove or leave in the sample json output, what do each of you think?

**Comment by ajbozarth** [comment]
realized I pushed my latest change to the wrong pr branch, fixed now

File: README.md (hunk 1 above)
Comment (stevemart) [reply README.md]: @ajbozarth i think it's fine to include the sample output

Comment (ajbozarth) [reply README.md]: thanks @stevemart 
Once @MLnick gives the ok on this I'll merge it
Though I do have one issue, does the new line between the curl and it's output break the flow of understanding? It just feels off to me. Should we move that note after the output or maybe somewhere before the curl command? Or should we add a line specifying that the output block is what output may look like?

File: README.md [hunk 5]
```diff
@@ -128,31 +127,18 @@ The model will only be available internally, but can be accessed externally thro
...
-    cd MAX-Image-Caption-Generator
-
-To build the docker image locally, run:
+To run the docker image, which automatically starts the model serving API, run:
 
-    docker build -t max-im2txt .
+    docker run -it -p 5000:5000 codait/max-image-caption-generator
 
-All required model assets will be downloaded during the build process.
+This will pull the latest docker image available on Docker Hub to run. If you'd rather build the model locally you can
```
Comment (bdwyer2) [inline README.md]: Maybe I'm being to nit picky but that command will not always pull the latest version from docker hub. If the user has pulled the image in the past and it is cached locally docker will run that image even if a newer image is available on docker hub.

Comment (ajbozarth) [reply README.md]: that is correct, maybe I should reword it to clarify that it pulls the latest the first time then reuses that same version from then on. Maybe add `...run, or use a previous copy if you've run the command previously.` what do you think?

Comment (bdwyer2) [reply README.md]: I really like the simplicity of being able to use the model with a single command and would hate to lose that.

Maybe something like

> This will pull a pre-built image from docker hub (or use an existing image if already cached locally) and run it.

Comment (ajbozarth) [reply README.md]: ooo, I like that, I'll update to use it

Comment (ajbozarth) [reply README.md]: updated

File: README.md (hunk 1 above)
Comment (MLnick) [reply README.md]: I like the wording of the paragraph now, but I agree the placement looks odd. I suggest moving the paragraph just above the line "You can also test it on the command line, for example:"

File: README.md [hunk 6]
```diff
@@ -232,15 +222,23 @@ network stack. This is done in the following steps:
...
+    docker run -it -p 5000:5000 -p 8088:8088 --name max-image-caption-generator codait/max-image-caption-generator
 
 Build the web app image by running:
 
     docker build -t webapp .
 
 Run the web app container using:
 
-    docker run --net='container:max-im2txt' -it webapp
+    docker run --net='container:max-image-caption-generator' -it webapp
```
Comment (MLnick) [inline README.md]: Sorry for nitpicking some more, but really we should also change the tag `webapp` here -> `max-image-caption-generator-web-app` for consistency throughout

**Comment by MLnick** [comment]
A couple further minor comments, pretty close now

File: README.md (hunk 6 above)
Comment (ajbozarth) [reply README.md]: good idea I'll update

File: README.md (hunk 1 above)
Comment (ajbozarth) [reply README.md]: I'll try moving it

Comment (ajbozarth) [reply README.md]: I moved it and it looks better there as you thought.

File: README.md (hunk 6 above)
Comment (ajbozarth) [reply README.md]: done

**Comment by ajbozarth** [comment]
@MLnick Made those requested updates

File: README.md [hunk 7]
```diff
@@ -232,15 +222,23 @@ network stack. This is done in the following steps:
 Modify the command that runs the Image Caption Generator REST endpoint to map an additional port in the container to a
 port on the host machine. In the example below it is mapped to port `8088` on the host but other ports can also be used.
 
-    docker run -it -p 5000:5000 -p 8088:8088 --name max-im2txt max-im2txt
+    docker run -it -p 5000:5000 -p 8088:8088 --name max-image-caption-generator codait/max-image-caption-generator
 
 Build the web app image by running:
 
-    docker build -t webapp .
+    docker build -t web-app .
```
Comment (MLnick) [inline README.md]: I actually meant the docker image tag should be `max-image-caption-generator-web-app` per my previous comment. Looks like this was updated to just `web-app`?

Comment (ajbozarth) [reply README.md]: hmmm, I guess we can do that, even if it's really wordy for a local image it would be clearer

Comment (ajbozarth) [reply README.md]: updated, also in retrospect this makes sense since I plan to update the models in a similar fashion next
